src/frontend/components/pages/shopping-list.component.ts

```diff
@@ -1,8 +1,54 @@
-import {Component} from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {ShoppingListService} from "./shopping-list.service";
+import {Group} from "../../models/Group";
+import {Article} from "../../models/Article";
+import {ChipComponent} from "../elements/chip/chip.component";
+import {CardComponent} from "../elements/card/card.component";
+import {AddComponent} from "../elements/add/add.component";
 
 @Component({
     templateUrl: 'frontend/components/pages/shopping-list.component.html',
-    styleUrls: ['frontend/components/pages/shopping-list.component.css']
+    styleUrls: ['frontend/components/pages/shopping-list.component.css'],
+    providers: [ShoppingListService],
+    directives: [CardComponent, ChipComponent, AddComponent]
+
 })
-export class ShoppingListComponent {
+export class ShoppingListComponent implements OnInit {
+    private groups: Group[] = [];
+    private articles: Article[] = [];
+    private addText: string = 'Add article';
+
+
+    constructor(private shoppingListService: ShoppingListService) {
+    }
+
+    getArticlesForGroup(id: number) {
```

Comment:
Return-Typ in Methoden-Signatur angeben: 
`getArticlesForGroup(id: number) : Article[] { ... }`